python-checks/src/main/java/org/sonar/python/checks/cdk/UnrestrictedAdministrationCheck.java

```diff
@@ -0,0 +1,225 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.cdk;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.DictionaryLiteral;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.ListLiteral;
+import org.sonar.plugins.python.api.tree.NumericLiteral;
+import org.sonar.plugins.python.api.tree.StringLiteral;
+
+import static org.sonar.python.checks.cdk.CdkPredicate.isListLiteral;
+import static org.sonar.python.checks.cdk.CdkPredicate.isNumericLiteral;
+import static org.sonar.python.checks.cdk.CdkPredicate.isString;
+import static org.sonar.python.checks.cdk.CdkPredicate.isStringLiteral;
+import static org.sonar.python.checks.cdk.CdkUtils.getArgument;
+import static org.sonar.python.checks.cdk.CdkUtils.getCall;
+import static org.sonar.python.checks.cdk.CdkUtils.getDictionary;
+
+@Rule(key="S6321")
+public class UnrestrictedAdministrationCheck extends AbstractCdkResourceCheck {
+  private static final String MESSAGE = "Change this IP range to a subset of trusted IP addresses.";
+
+  private static final String IP_PROTOCOL = "ip_protocol";
+  private static final String CIDR_IP = "cidr_ip";
+  private static final String CIDR_IPV6 = "cidr_ipv6";
+  private static final String IPPROTOCOL = "ipProtocol";
+  private static final String CIDRIP = "cidrIp";
+  private static final String CIDRIPV6 = "cidrIpv6";
+
+
+  private static final Set<String> SENSITIVE_PROTOCOL = Set.of("tcp", "6");
+  private static final String ANY_PROTOCOL = "-1";
+  private static final String EMPTY_IPV4 = "0.0.0.0/0";
+  private static final String EMPTY_IPV6 = "::/0";
+  private static final long[] ADMIN_PORTS = new long[]{22, 3389};
+
+  @Override
+  protected void registerFqnConsumer() {
+    checkFqn("aws_cdk.aws_ec2.CfnSecurityGroup", UnrestrictedAdministrationCheck::checkCfnSecurityGroup);
+    checkFqn("aws_cdk.aws_ec2.CfnSecurityGroupIngress", UnrestrictedAdministrationCheck::checkCfnSecurityGroupIngress);
+  }
+
+  // Checks methods
+  private static void checkCfnSecurityGroup(SubscriptionContext ctx, CallExpression callExpression) {
+    getArgument(ctx, callExpression, "security_group_ingress")
+      .flatMap(flow -> flow.getExpression(isListLiteral()).map(ListLiteral.class::cast))
```

Comment:
Maybe move this logic to a method like `Optional<ListLiteral> getListExpression(ExpressionFlow expression)`